audiblelight/augmentation.py

```
@@ -0,0 +1,1716 @@
+    of 8kHz (required by the GSM Full Rate codec), although the quality of the resampling algorithm can be specified.
+
+    Arguments:
+        sample_rate (utils.Numeric): the sample rate for the effect to use.
+        quality: the quality of the resampling. By default, will be sampled between 0 and 3 (inclusive).
+    """
+
+    # Don't use the highest resampling quality (4) as it is much slower than the others
+    QUALITIES = range(4)
+
+    def __init__(
+        self,
+        sample_rate: utils.Numeric = utils.SAMPLE_RATE,
+        quality: Optional[Union[utils.Numeric, utils.DistributionLike]] = None,
+    ):
+        super().__init__(sample_rate)
+        self.quality = int(
+            utils.sanitise_positive_number(
+                self.sample_value(quality, lambda: np.random.choice(self.QUALITIES))
+            )
+        )
+        self.params = dict(quality=self.quality)
+
+        from pedalboard import GSMFullRateCompressor as PBGSMFullRateCompressor
+        from pedalboard import Resample
+
+        self.fx = PBGSMFullRateCompressor(quality=Resample.Quality(self.quality))
+
+
+class MP3Compressor(EventAugmentation):
+    """
+    Applies the LAME MP3 encoder in real-time to add compression artifacts to the audio stream.
+
+    Currently only supports variable bit-rate mode (VBR) and accepts a floating-point VBR quality value
+    (between 0.0 and 10.0; lower is better). Note that the MP3 format only supports 8kHz, 11025Hz, 12kHz, 16kHz,
+    22050Hz, 24kHz, 32kHz, 44.1kHz, and 48kHz audio; if an unsupported sample rate is provided, an exception will be
+    thrown at processing time.
+
+    Arguments:
+        sample_rate (utils.Numeric): the sample rate for the effect to use.
+        vbr_quality: the quality of the resampling. By default, will be sampled between 2 and 10.
+    """
+
+    VBR_MIN, VBR_MAX = 2.001, 9.999
+
+    def __init__(
+        self,
+        sample_rate: utils.Numeric = utils.SAMPLE_RATE,
```

should we say something here if the global sampling rate (`utils.SAMPLE_RATE`) is not supported?

---

yep, we should probably raise an error straight when the class is initialized if the sample rate is wrong